src/AISmart.GAgent.Twitter/Agent/TwitterGAgent.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using AISmart.Agent.GEvents;
+using AISmart.Agents;
+using AISmart.GAgent.Core;
+using AISmart.GEvents.Social;
+using AISmart.GEvents.Twitter;
+using AISmart.Grains;
+using Microsoft.Extensions.Logging;
+using Orleans.Providers;
+
+namespace AISmart.Agent;
+
+[Description("Handle telegram")]
+[StorageProvider(ProviderName = "PubSubStore")]
+[LogConsistencyProvider(ProviderName = "LogStorage")]
+public class TwitterGAgent : GAgentBase<TwitterGAgentState, TweetGEvent>, ITwitterGAgent
+{
+    private readonly ILogger<TwitterGAgent> _logger;
+
+    public TwitterGAgent(ILogger<TwitterGAgent> logger) : base(logger)
+    {
+        _logger = logger;
+    }
+    
+    public override Task<string> GetDescriptionAsync()
+    {
+        return Task.FromResult(
+            "Represents an agent responsible for informing other agents when a twitter thread is published.");
+    }
+    
+    [EventHandler]
+    public async Task HandleEventAsync(ReceiveReplyEvent @event)
+    {
+        _logger.LogInformation("Tweet ReceiveReplyEvent " + @event.TweetId);
+        
+        RaiseEvent(new ReceiveTweetReplyGEvent
+        {
+            TweetId = @event.TweetId,
+            Text = @event.Text
+        });
+        await ConfirmEvents();
+        await PublishAsync(new SocialEvent()
+        {
+            Content = @event.Text,
+            TweetId = @event.TweetId
```

Comment:
Why is there a need for anyone receiving socialevent to know about the tweetid? This should not be a field in socialevent